packages/data-table/src/mdc-data-table.ts

```diff
@@ -147,6 +150,17 @@ export class MdcDataTable extends MdcComponent<MDCDataTableFoundation> implement
 
   handleNavigationClick(type: 'first' | 'prev' | 'next' | 'last') {
     this.emit(NAVIGATION_EVENT, { type }, true);
+
```

Comment:
Yeah i've changed my mind about this several times. So my thinking is either...

1 - the component needs to own the navigation and therefore manage the activePage property itself which the consumer can subscribe to.
2 - the responsibility of managing the activePage property falls to the consumer itself and can hook into the pagination events to do so.

Number 1 seems the most efficient way but your concern above about "what if an error occurred when loading the next page in consumer code" is completely valid and not easily resolved.

Number 2 resolves your concern as the consumer can manage the activePage property after successfully loading the next block of data. However, we are likely to have the same code block to subscribe to the navigation events and manage the activePage in every consumer use of this component. Ok, it can be wrapped in a common component and abstracted away etc but seems wrong somehow.

I'll take your steer on this before making any changes.

Reply:
You can make this optional, not handling the events by default. Those who need it will turn on.